packages/@romefrontend/compiler/lint/rules/js/noUnusedTemplateLiteral.ts

```diff
@@ -0,0 +1,35 @@
+import {Path, TransformExitResult} from "@romefrontend/compiler";
+import {descriptions} from "@romefrontend/diagnostics";
+import {JSTemplateElement, jsStringLiteral} from "@romefrontend/ast";
+
+function containsSpecialCharacters(node: JSTemplateElement): boolean {
+	return /['"\n]/.test(node.raw);
+}
+
+export default {
+	name: "jsNoUnusedTemplateLiteral",
+	enter(path: Path): TransformExitResult {
+		const {context, node} = path;
+
+		if (
+			node.type === "JSTemplateLiteral" &&
+			!path.findAncestry((path) =>
+				path.node.type === "JSTaggedTemplateExpression"
+			) &&
```

Comment:
You can just check the parent. Doing this would allow:

```
foo`${`i am redundant`}`;
```

```suggestion
			path.parent.type !== "JSTaggedTemplateExpression" &&
```